meta-resin-common/classes/image_types_resin.bbclass

```diff
@@ -175,60 +183,99 @@ IMAGE_CMD_resinos-img () {
     #
 
     dd if=/dev/zero of=${RESIN_RAW_IMG} bs=1024 count=0 seek=${RESIN_RAW_IMG_SIZE}
-    parted -s ${RESIN_RAW_IMG} mklabel msdos
+
+    if [ "${PARTITION_TABLE_TYPE}" = "msdos" ]; then
```

Comment:
This seems to be specific to the first partition. It should be after `# resin-boot` as you did it for the next partitions.